common/unsafe/src/main/java/org/apache/spark/unsafe/types/ByteArray.java

```diff
@@ -160,6 +160,39 @@ public static byte[] concat(byte[]... inputs) {
     return result;
   }
 
+  public static byte[] concatWS(byte[] delimiter, byte[]... inputs) {
+    // Compute the total length of the result
+    long totalLength = 0;
+    for (byte[] input : inputs) {
+      if (input != null) {
+        totalLength += input.length + delimiter.length;
+      } else {
+        return null;
+      }
+    }
+    if (totalLength > 0) totalLength -= delimiter.length;
+    // Allocate a new byte array, and copy the inputs one by one into it
+    final byte[] result = new byte[Ints.checkedCast(totalLength)];
+    int offset = 0;
+    for (int i = 0; i < inputs.length; i++) {
+      byte[] input = inputs[i];
+      int len = input.length;
+      Platform.copyMemory(
```

Comment:
this seems copied from L154 above, please dedup into one place?

Reply:
I didn't want to accidentally change existing behavior or performance so I thought a little copy-paste was justified in this isolated code. But I probably concern too much)
Removed